### PR TITLE
Further facet improvements.

### PR DESCRIPTION
**Pull request by claygoddard**
Add support for unique explicit terms limit.
Add support for pruning merged buckets when writing iterative state.

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/search/facet/FacetParser.java [hunk 1]
```diff
@@ -546,6 +546,7 @@ public FacetField parse(Object arg) throws SyntaxError {
         facet.allBuckets = getBoolean(m, "allBuckets", facet.allBuckets);
         facet.method = FacetField.FacetMethod.fromString(getString(m, "method", null));
         facet.cacheDf = (int)getLong(m, "cacheDf", facet.cacheDf);
+        facet.merged_buckets_Limit = (int)getLong(m, "merged_buckets_limit", facet.merged_buckets_Limit);
```
Comment (claygoddard) [inline solr/core/src/java/org/apache/solr/search/facet/FacetParser.java]: fix casing

Comment (claygoddard) [reply solr/core/src/java/org/apache/solr/search/facet/FacetParser.java]: or go back to camel case

File: solr/core/src/java/org/apache/solr/search/facet/UniqueAgg.java [hunk 2]
```diff
@@ -202,12 +208,10 @@ public Object getShardValue(int slot) throws IOException {
...
       SimpleOrderedMap map = new SimpleOrderedMap();
       map.add("unique", unique);
-
-      int maxExplicit=100;
-      // TODO: make configurable
+      
       // TODO: share values across buckets
-      if (unique <= maxExplicit) {
-        List lst = new ArrayList( Math.min(unique, maxExplicit) );
+      if (unique <= numValsExplicit && numValsExplicit > 0) {
```
Comment (claygoddard) [inline solr/core/src/java/org/apache/solr/search/facet/UniqueAgg.java:213]: remove second part of `&&` expression

**Review by bradbesserman — APPROVED** [review]
LGTM!

**Pull request merged**